lib/utils/mlock/mlock_unsupported.go

```diff
@@ -0,0 +1,28 @@
+//go:build !linux
+
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mlock
+
+import (
+	"github.com/gravitational/trace"
+)
+
+func lockMemory() error {
+	// Mlockall is only available on linux.
+	return trace.NotImplemented("not implemented")
```

Comment:
TBH, I'd return `nil` here just to keep it simple. Otherwise, you need to call it and check for the error, knowing that MacOS and Windows always return it.